File: C++ ad y ad/3_ Orientado a objetos/S01.3_ Automovil/Solucion_ejercicio_automovil/Ejercicio_automovil/automovil.cpp
#include "automovil.h"


Automovil::Automovil(string color)
{
	this->color = color;

	estaPrendido = false;
	estaEnMovimiento = false;
}
Automovil::~Automovil()
{
	cout << "El auto " << color << " fue liberado de la memoria.\n\n";
}


void Automovil::prender()
{
	if (!estaPrendido)
	{
		estaPrendido = true;

		cout << "El auto se prendio. Esta listo para empezar a andar!\n";
	}
	else
	{
		cout << "El auto ya esta prendido.\n";
	}
}
void Automovil::apagar()
{
	if (estaPrendido)
	{
		if (estaEnMovimiento)
		{
			cout << "El auto debe estar detenido para poder apagarse.\n";
		}
		else
		{
			estaPrendido = false;

			cout << "El auto apago sus motores.\n";
		}
	}
	else
	{
		cout << "El auto ya esta apagado.\n";
	}
}

void Automovil::acelerar()
{
	if (estaPrendido)
	{
		if (!estaEnMovimiento)
		{
			estaEnMovimiento = true;

			cout << "El auto acelero y empezo a recorrer su camino.\n";
		}
		else
		{
			cout << "El auto ya acelero!\n";
		}
	}
	else
	{
		cout << "Para acelerar, el auto debe prenderse.\n";
	}
}
void Automovil::frenar()
{
	if (estaPrendido)
	{
		if (estaEnMovimiento)
		{
			estaEnMovimiento = false;

			cout << "El auto detuvo su marcha lentamente hasta quedarse quieto.\n";
		}
		else
		{
			cout << "El auto ya esta quieto.\n";
		}
	}
	else
	{
		cout << "Para poder frenar, el auto debe estar prendido y en marcha.\n";
	}
}

void Automovil::mostrarDatos()
{
	cout << "Color del auto: " << color << ".\n";
	cout << "Estado del auto: ";
	
	if (estaPrendido)
	{
		cout << "Prendido y ";

		if (estaEnMovimiento)
		{
			cout << "andando.\n";
		}
		else
		{
			cout << "quieto.\n";
		}
	}
	else
	{
		cout << "Apagado.\n";
	}
}

File: C# ad y ad/2_ Orientado a objetos/S01.3_ Automovil/Solucion_ejercicio_automovil/Ejercicio_automovil/Automovil.cs
using System;

namespace Ejercicio_automovil
{
	class Automovil
	{
		string color;
		bool estaPrendido;
		bool estaEnMovimiento;


		public Automovil(string color)
		{
			this.color = color;

			estaPrendido = false;
			estaEnMovimiento = false;
		}


		public void Prender()
		{
			if (!estaPrendido)
			{
				estaPrendido = true;

				Console.WriteLine("El auto se prendió. ¡Está listo para empezar a andar!");
			}
			else
			{
				Console.WriteLine("El auto ya está prendido.");
			}
		}
		public void Apagar()
		{
			if (estaPrendido)
			{
				if (estaEnMovimiento)
				{
					Console.WriteLine("El auto debe estar detenido para poder apagarse.");
				}
				else
				{
					estaPrendido = false;

					Console.WriteLine("El auto apagó sus motores.");
				}
			}
			else
			{
				Console.WriteLine("El auto ya está apagado.");
			}
		}

		public void Acelerar()
		{
			if (estaPrendido)
			{
				if (!estaEnMovimiento)
				{
					estaEnMovimiento = true;

					Console.WriteLine("El auto aceleró y empezó a recorrer su camino.");
				}
				else
				{
					Console.WriteLine("¡El auto ya aceleró!");
				}
			}
			else
			{
				Console.WriteLine("Para acelerar, el auto debe prenderse.");
			}
		}
		public void Frenar()
		{
			if (estaPrendido)
			{
				if (estaEnMovimiento)
				{
					estaEnMovimiento = false;

					Console.WriteLine("El auto detuvo su marcha lentamente hasta quedarse quieto.");
				}
				else
				{
					Console.WriteLine("El auto ya está quieto.");
				}
			}
			else
			{
				Console.WriteLine("Para poder frenar, el auto debe estar prendido y en marcha.");
			}
		}

		public void MostrarDatos()
		{
			Console.WriteLine("Color del auto: " + color + ".");
			Console.Write("Estado del auto: ");

			if (estaPrendido)
			{
				Console.Write("Prendido y ");

				if (estaEnMovimiento)
				{
					Console.WriteLine("andando.");
				}
				else
				{
					Console.WriteLine("quieto.");
				}
			}
			else
			{
				Console.WriteLine("Apagado.");
			}
		}

	}
}
